diff -Nur rp-pppoe-3.5-orig/src/if.c rp-pppoe-3.5/src/if.c
--- rp-pppoe-3.5-orig/src/if.c	Mon Jul  8 16:38:24 2002
+++ rp-pppoe-3.5/src/if.c	Wed Sep  4 12:04:11 2002
@@ -533,7 +533,7 @@
 	return -1;
     }
 #elif defined(HAVE_STRUCT_SOCKADDR_LL)
-    if (send(sock, pkt, size, 0) < 0) {
+    if ((send(sock, pkt, size, 0) < 0) && ( errno != ENOBUFS )) {
 	sysErr("send (sendPacket)");
 	return -1;
     }
